### PR TITLE
Add code to respect tag fire limit on preview based on container setting, #PG-3535

### PR DESCRIPTION
**Pull request by AltamashShaikh**
### Description:

Add code to respect tag fire limit on preview based on container setting,
Fixes: #PG-3535, #801

### Review

* [ ] [Functional review done](https://developer.matomo.org/guides/pull-request-reviews#functional-review-done)
* [ ] [Potential edge cases thought about](https://developer.matomo.org/guides/pull-request-reviews#potential-edge-cases-thought-about) (behavior of the code with strange input, with strange internal state or possible interactions with other Matomo subsystems)
* [ ] [Usability review done](https://developer.matomo.org/guides/pull-request-reviews#usability-review-done) (is anything maybe unclear or think about anything that would cause people to reach out to support)
* [ ] [Security review done](https://developer.matomo.org/guides/security-in-piwik#checklist)
* [ ] [Wording review done](https://developer.matomo.org/guides/pull-request-reviews#translations-wording-review-done)
* [ ] [Code review done](https://developer.matomo.org/guides/pull-request-reviews#code-review-done)
* [ ] [Tests were added if useful/possible](https://developer.matomo.org/guides/pull-request-reviews#tests-were-added-if-usefulpossible)
* [ ] [Reviewed for breaking changes](https://developer.matomo.org/guides/pull-request-reviews#reviewed-for-breaking-changes)
* [ ] [Developer changelog updated if needed](https://developer.matomo.org/guides/pull-request-reviews#developer-changelog-updated-if-needed)
* [ ] [Documentation added if needed](https://developer.matomo.org/guides/pull-request-reviews#documentation-added-if-needed)
* [ ] Existing documentation updated if needed

### CODE REVIEW

File: Updates/5.2.0-b1.php [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+use Piwik\Plugins\TagManager\Template\Tag\MatomoTag;
+use Piwik\Plugins\TagManager\UpdateHelper\NewTagParameterMigrator;
+use Piwik\Updater;
+use Piwik\Updater\Migration;
+use Piwik\Updater\Migration\Factory as MigrationFactory;
+use Piwik\Updates as PiwikUpdates;
+
+/**
+ * Update for version 5.0.0-rc8.
```
Comment (snake14) [inline Updates/5.2.0-b1.php]: Not a big deal, but the comment appears to be incorrect.

File: lang/en.json [hunk 2]
```diff
@@ -1056,6 +1056,8 @@
         "ResumeX": "Resume %s",
         "ResumeTagConfirm": "Are you sure you want to activate this tag again?",
         "DiffPaused": "Paused",
-        "DiffAddedPaused": "Added in paused state"
+        "DiffAddedPaused": "Added in paused state",
+        "TagFireLimitAllowedInPreviewModeTitle": "Allow tag fire limits to work in preview mode",
+        "TagFireLimitAllowedInPreviewModeDescription": "Description"
```
Comment (snake14) [inline lang/en.json]: Is this still waiting on the copy?

File: tests/UI/Container_spec.js [hunk 3]
```diff
@@ -95,7 +95,22 @@ describe("Container", function () {
...
     });
 
+    it('should set isTagFireLimitAllowedInPreviewMode as active when set and save', async function(){
+        await page.evaluate(() => $('#isTagFireLimitAllowedInPreviewMode').click());
+        await page.waitForTimeout(250);
+        await page.waitForNetworkIdle();
+        await capture.page(page, 'edit_url_with_isTagFireLimitAllowedInPreviewMode');
+    });
+
+    it('should set isTagFireLimitAllowedInPreviewMode as active when set and save', async function(){
```
Comment (snake14) [inline tests/UI/Container_spec.js]: Should this say `inactive` instead of `active`?

**Review by snake14 — APPROVED** [review]
Looks good! I just left a few comments.

Edit: Tested locally. The migration ran without issue and I confirmed that when the new setting is enabled, the tag triggers based on the tag's Fire limit configuration.

File: tests/UI/Container_spec.js (hunk 3 above)
Comment (AltamashShaikh) [reply tests/UI/Container_spec.js]: yes :+1:

File: lang/en.json (hunk 2 above)
Comment (AltamashShaikh) [reply lang/en.json]: Just updated the copies

File: Updates/5.2.0-b1.php (hunk 1 above)
Comment (AltamashShaikh) [reply Updates/5.2.0-b1.php]: Fixed :+1:

**Review by snake14 — APPROVED** [review]

**Pull request merged**